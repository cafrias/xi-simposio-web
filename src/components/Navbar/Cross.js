import React from 'react'

function Cross({style}) {
  const {x} = style
  return (
    <svg style={{transform: `scale(${x}, ${x})`}} xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
      <g fill="none" fillRule="evenodd" stroke="#BCBCBC" strokeWidth="2" transform="translate(13 12)" strokeLinecap="square">
        <path d="M.5 21.5L22.5.5M.5.5L22.5 21.5"/>
      </g>
    </svg>
  )
}

export default Cross
