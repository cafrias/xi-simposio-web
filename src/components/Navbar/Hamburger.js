import React from 'react'

function Hamburger({style}) {
  const {x} = style
  return (
    <svg style={{transform: `scale(${x}, ${x})`}} xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
      <g fill="none" fillRule="evenodd" stroke="#9B9B9B" strokeWidth="2" transform="translate(12 15)" strokeLinecap="square">
        <path d="M0 1L24 1M0 9L24 9M0 17L24 17"/>
      </g>
    </svg>
  )
}

export default Hamburger
