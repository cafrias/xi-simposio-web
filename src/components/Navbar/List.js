import React from 'react'
import {StaggeredMotion, spring} from 'react-motion'

const MENU_ITEMS = [
  {
    name: '¿Qué es el simposio?',
    link: '#about'
  },
  {
    name: 'Disertantes',
    link: '#speakers'
  },
  {
    name: 'Cronograma',
    link: '#schedule'
  },
  {
    name: 'Inscripción',
    link: '#admission'
  },
  {
    name: 'Ubicación',
    link: '#place'
  }
]

const STIFFNESS = 210
const DAMPING = 18

function TopList() {
  return (

    <ul className="nav__list top-nav__list">
      {
        MENU_ITEMS.map(({link, name}, i) => (
          <li key={i}><a href={link} className="top-nav__link">{name}</a></li>
        ))
      }
    </ul>
  )
}

function List({topNav, closeAndNavigate}) {
  if(topNav) return TopList()

  return (
    <StaggeredMotion
      defaultStyles={MENU_ITEMS.map((item, i) => ({x: -100, o: 0}))}
      styles={prevInterpolatedStyles => prevInterpolatedStyles.map((_, i) => {
        return i === 0
          // Initial stiffness and damping
          ? {
              x: spring(0, {stiffness: STIFFNESS, damping: DAMPING}),
              o: spring(1)
            }
          // Final stiffness and damping
          : {
              x: spring(prevInterpolatedStyles[i - 1].x, {
                stiffness: STIFFNESS, damping: DAMPING
              }),
              o: spring(prevInterpolatedStyles[i - 1].o)
            };
      })}
    >
      {interpolatingStyles =>
        <ul className="nav__list">
          {interpolatingStyles.map(({x, o}, i) => {
            const styles = {
              transform: `translate(${x}px, 0)`,
              opacity: o
            }

            return (
              <li key={i} style={styles}
                onClick={() => closeAndNavigate(MENU_ITEMS[i].link)}>
                <a href={MENU_ITEMS[i].link} className="top-nav__link">
                  {MENU_ITEMS[i].name}
                </a>
              </li>
            )
          })}
        </ul>
      }
    </StaggeredMotion>
  )
}

export default List
