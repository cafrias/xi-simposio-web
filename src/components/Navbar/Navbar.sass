@import '../../_sass/_breakpoints.sass'
@import '../../_sass/_constants.sass'

.top-nav
  font-weight: normal
  -webkit-font-smoothing: antialiased
  width: 100%
  display: flex
  flex-wrap: wrap
  align-items: center
  justify-content: center
  position: fixed
  z-index: 10
  top: 0
  left: 0

.top-nav--open
  height: 100vh
  align-items: center
  justify-content: flex-start
  flex-direction: column
  background-color: #000
  transition: background-color .3s

.top-nav__on-top
  width: 100%
  display: flex
  flex-wrap: wrap
  align-items: center
  justify-content: space-between
  padding: 0 .75rem
  height: 4.236rem

.subline
  height: 1px
  width: 85%
  background-color: $light-grey

// HAMBURGER ___________________________________________________________________
.hamburger
  display: flex
  justify-content: flex-end
  width: 15%

// LIST ________________________________________________________________________
.nav__list
  display: flex
  flex-direction: column
  list-style: none
  padding: 0
  justify-content: space-around
  flex-grow: 2
  width: 100%
  margin-top: 0
  margin-bottom: 3.272rem
  text-align: left
  padding-left: 3.272rem

.top-nav__list
  display: none

// LINK ________________________________________________________________________
.top-nav__link
  text-transform: uppercase
  font-size: 1.618rem
  color: $dark-grey
  text-decoration: none
  display: block
  width: 100%

.top-nav__link:hover
  color: $primary
  transition: .5s

// LOGO ________________________________________________________________________
.logo
  width: 3.272rem

.logo-container
  display: flex
  align-items: center
  width: 75%

// DESKTOP _____________________________________________________________________
+for-desktop-up
  .top-nav
    height: 6.854rem
    width: 100%
    padding: 0 4.236rem

  .top-nav__on-top
    align-items: center
    flex-wrap: nowrap

  .subline
    height: 2px

  // LIST
  .top-nav__list
    display: flex
    flex-grow: 2
    flex-direction: row
    justify-content: space-between
    margin-bottom: 0

  // LOGO
  .logo
    width: 5.295rem

  .logo-container
    width: 25%

  // HAMBURGER
  .hamburger
    display: none

// DESKTOP BIG BREAKPOINT ______________________________________________________
+for-big-desktop-up
  .top-nav
    padding: 0 11.089rem
