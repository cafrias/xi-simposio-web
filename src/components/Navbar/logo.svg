<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 196.41 143"><defs><style>.a{fill:#c25455;}.b{fill:#f2f2f2;}</style></defs><title>logo</title><path class="a" d="M212.13,123.18s3,51.46-53.28,56.6l-14.5.51V95.72L128.18,121V194.5h29.55c71-4.71,69.62-71.32,69.62-71.32s2.3-54.81-50.5-68.42a10.07,10.07,0,0,0-2-.52c-2.41-.39-.06-.36-2.67-.6h0l-8.81,13.47h0c1.57.14,7.36,1.43,8.81,1.8,42.09,10.69,40,54.27,40,54.27" transform="translate(-30.95 -51.5)"/><path class="b" d="M31,51.51H159.15s.91-.05,4.67.15a14.67,14.67,0,0,1,3.89.69l-8.94,13.88H100.54V95.37h39.94l-13.6,21.14-7.48-7H99.7v84.65l-14.78-8.37V109.5H59L50.78,95.37H84.91V66.23h-46Z" transform="translate(-30.95 -51.5)"/></svg>