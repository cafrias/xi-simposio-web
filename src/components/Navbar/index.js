import React, {Component} from 'react'
import {Motion, spring, presets} from 'react-motion'

// IMPORT ASSETS _______________________________________________________________
import './Navbar.sass'
import logo from './logo.svg'
import Hamburger from './Hamburger'
import Cross from './Cross'

// ANIMATIONS __________________________________________________________________
import FadeIn from '../../animations/FadeIn'

// IMPORT COMPONENTS ___________________________________________________________
import List from './List'

class Navbar extends Component {
  constructor(props) {
    super(props)

    this.state = {
      navOpen: false
    }
  }

  toggleMenuHandler() {
    this.setState({
      navOpen: !this.state.navOpen
    })
  }

 addMotionToComponent(component, key) {
   return (
     <Motion key={key} defaultStyle={{x: 0}}
       style={{x: spring(1, presets.gentle)}}
       >
       {interpolatingStyle => component({style: interpolatingStyle})}
     </Motion>
   )
 }

  render() {
    const {navOpen} = this.state
    const {
      addMotionToComponent
    } = this
    return (
      <FadeIn
        onRest={f => f}
      >
        <header role="banner"
          className={`top-nav ${navOpen ? 'top-nav--open' : ''}`}
          >
          <div className="top-nav__on-top">
            <div className="logo-container">
              <img src={logo} alt="logo" className="logo" />
            </div>
            <div className="hamburger" onClick={this.toggleMenuHandler.bind(this)}>
              {
                navOpen
                ? addMotionToComponent(Cross, 1)
                : addMotionToComponent(Hamburger, 2)
              }
            </div>
            <List topNav={true} />
          </div>
          {
            navOpen
            ? <List topNav={false} />
            : null
          }
        </header>
      </FadeIn>
    )
  }
}

export default Navbar
