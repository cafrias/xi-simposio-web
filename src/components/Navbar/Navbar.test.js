import React from 'react'
import {shallow} from 'enzyme'

import Navbar from './index'

describe('Navbar', () => {
  var navbarWrapper

  beforeEach(() => navbarWrapper = shallow(<Navbar />))

  it('should render without errors', () => {
    expect(navbarWrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(navbarWrapper).toMatchSnapshot()
  })
})
