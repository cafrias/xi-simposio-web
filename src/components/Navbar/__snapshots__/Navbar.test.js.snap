exports[`Navbar should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <FadeIn
    onRest={[Function]}>
    <header
        className="top-nav "
        role="banner">
        <div
            className="top-nav__on-top">
            <div
                className="logo-container">
                <img
                    alt="logo"
                    className="logo"
                    src="logo.svg" />
            </div>
            <div
                className="hamburger"
                onClick={[Function]}>
                <Motion
                    defaultStyle={
                        Object {
                            "x": 0,
                          }
                    }
                    style={
                        Object {
                            "x": Object {
                              "damping": 14,
                              "precision": 0.01,
                              "stiffness": 120,
                              "val": 1,
                            },
                          }
                    }>
                    [Function]
                </Motion>
            </div>
            <List
                topNav={true} />
        </div>
    </header>
</FadeIn>,
  "nodes": Array [
    <FadeIn
      onRest={[Function]}>
      <header
            className="top-nav "
            role="banner">
            <div
                  className="top-nav__on-top">
                  <div
                        className="logo-container">
                        <img
                              alt="logo"
                              className="logo"
                              src="logo.svg" />
                  </div>
                  <div
                        className="hamburger"
                        onClick={[Function]}>
                        <Motion
                              defaultStyle={
                                    Object {
                                          "x": 0,
                                        }
                              }
                              style={
                                    Object {
                                          "x": Object {
                                            "damping": 14,
                                            "precision": 0.01,
                                            "stiffness": 120,
                                            "val": 1,
                                          },
                                        }
                              }>
                              [Function]
                        </Motion>
                  </div>
                  <List
                        topNav={true} />
            </div>
      </header>
</FadeIn>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Navbar />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Navbar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "navOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <FadeIn
          onRest={[Function]}>
          <header
                    className="top-nav "
                    role="banner">
                    <div
                              className="top-nav__on-top">
                              <div
                                        className="logo-container">
                                        <img
                                                  alt="logo"
                                                  className="logo"
                                                  src="logo.svg" />
                              </div>
                              <div
                                        className="hamburger"
                                        onClick={[Function]}>
                                        <Motion
                                                  defaultStyle={
                                                            Object {
                                                                      "x": 0,
                                                                    }
                                                  }
                                                  style={
                                                            Object {
                                                                      "x": Object {
                                                                        "damping": 14,
                                                                        "precision": 0.01,
                                                                        "stiffness": 120,
                                                                        "val": 1,
                                                                      },
                                                                    }
                                                  }>
                                                  [Function]
                                        </Motion>
                              </div>
                              <List
                                        topNav={true} />
                    </div>
          </header>
</FadeIn>,
        "_debugID": 4,
        "_renderedOutput": <FadeIn
          onRest={[Function]}>
          <header
                    className="top-nav "
                    role="banner">
                    <div
                              className="top-nav__on-top">
                              <div
                                        className="logo-container">
                                        <img
                                                  alt="logo"
                                                  className="logo"
                                                  src="logo.svg" />
                              </div>
                              <div
                                        className="hamburger"
                                        onClick={[Function]}>
                                        <Motion
                                                  defaultStyle={
                                                            Object {
                                                                      "x": 0,
                                                                    }
                                                  }
                                                  style={
                                                            Object {
                                                                      "x": Object {
                                                                        "damping": 14,
                                                                        "precision": 0.01,
                                                                        "stiffness": 120,
                                                                        "val": 1,
                                                                      },
                                                                    }
                                                  }>
                                                  [Function]
                                        </Motion>
                              </div>
                              <List
                                        topNav={true} />
                    </div>
          </header>
</FadeIn>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Navbar />,
}
`;
